### DISCUSSION TITLE
Responding with GZIP data

### DISCUSSION THREAD

**Discussion by jviotti** [uNetworking/uWebSockets#1627] [Q&A] [answered]
Does uWebSockets attempt to read `Accept-Encoding` by default and respond with GZIP automatically if it is compiled with ZLIB support, or are applications expected to handle this themselves?

**Comment by uNetworkingAB** [comment, marked as answer]
The server itself does not respond with compression for HTTP, but you could write a function that does. We could have HttpUtilities.h for such functions.

**Reply by jviotti** [reply]
Having utilities for this case would be great!

**Reply by ddevienne** [reply]
Note that if you are on a fast enough network, e.g. your Gb/s LAN, compression (especially with zlib) is almost always a loss.
I didn't test with HTTP, but I tested with WebSocket and partial compression of the message, and the bandwidth at which compression is a win is quite low / slow. LZ4 was a little better, i.e. a win at higher bandwidth, but not at LAN speed even. FWIW.
